src/main/c/functions.c

```diff
@@ -1,44 +1,159 @@
 #include <math.h>
 #include <stdlib.h>
+#include <string.h> // For memset.
 #include "data.h"
 #include "functions.h"
 
+
+#ifndef M_PI
+    #define M_PI 3.14159265358979323846
+#endif
+
 /**
  * Returns the size of a triangular matrix, without the diagonal.
  */
-size_t size_triangular_matrix(const size_t n) {
+size_t
+size_triangular_matrix(const size_t n) {
   return ((n - 1) * n) / 2;
 }
 
+/**
+ * Computes the distance between two particles.
+ */
+double
+compute_distance(const Particle *p1, const Particle *p2) {
+  const double x_diff = p1->x_coordinate - p2->x_coordinate;
+  const double y_diff = p1->y_coordinate - p2->y_coordinate;
+
+  return sqrt((x_diff * x_diff) + (y_diff * y_diff));
+}
+
 /**
  * Computes the overlap between two particles.
  * Note: If the overlap is negative, there is no overlap.
  */
-double compute_overlap(const Particle p1, const Particle p2) {
-  double d = p1.radious + p2.radious;
-
-  double x_diff = p1.x_coordinate - p2.x_coordinate;
-  double y_diff = p1.y_coordinate - p2.y_coordinate;
-  double distance = sqrt((x_diff * x_diff) + (y_diff * y_diff));
+double 
+compute_overlap(const Particle *p1, const Particle *p2) {
+  const double d = p1->radious + p2->radious;
+  const double distance = compute_distance(p1, p2);
 
   return d - distance;
 }
 
 /**
  * Takes two arrays of vectors, and sums together the vectors with the same index.
  */
-void sum_vectors(const size_t size, const Vector *v1, const Vector *v2, Vector *result) {
+void
+sum_vectors(const size_t size, const Vector *v1, const Vector *v2, Vector *result) {
   for (size_t i = 0; i < size; ++i) {
     result[i].x_component = v1[i].x_component + v2[i].x_component;
     result[i].y_component = v1[i].y_component + v2[i].y_component;
   }
 }
 
+/**
+ * Apply gravity to a vector of forces.
+ */
+void
+apply_gravity(const size_t size, const ParticleProperties *particles, Vector *forces) {
+  for (size_t i = 0; i < size; ++i) {
+    forces[i].y_component -= (particles[i].mass * 9.81d);
+  }
+}
+
+/**
+ * Compute the forces applied to P2 given it was collided by P1.
+ * Note: previous_normal and previous_tangent correspond to P2 with respect to P1.
+ */
+void
+collide_two_particles(const double dt, const Particle *p1, const Particle *p2, const Vector *velocity_p1, const Vector *velocity_p2, const ParticleProperties *properties_p2, Vector *force_p2, double *previous_normal, double *previous_tangent) {
```

Comment:
I strongly suggest to use lines no longer than 80 characters long. It is an awful experience having to scroll right and left to read a line of code. This is a GNU coding standard too, which is also compatible between most computer systems not only between really wide screen computer systems. https://www.gnu.org/prep/standards/standards.html#index-line-length.

Reply:
The lines had to be splitten I agree. But again, that is a change that should be done on another PR.

Finally, 80 chars is usually too short for modern programming languages, nowadays people suggest 100 and a occasionally maximum of 120.
Anyways, most of the lines that are too long re the function signatures, so not big deal to refactor them.

Reply:
> that is a change that should be done on another PR
I agree, but since they are now changed I think it would be OK to keep them this way, until we set our format constraints (on another PR).

Reply:
The thing is we have to sync this with master, and more changes mean more conflicts.

What about I revert this while syncing with master. And we discuss all these changes on a more detailed way on these days?
_(I will not do it if you prefer not)_.

Reply:
I agree, if it is simpler for you.